### DISCUSSION TITLE
No of points in point cloud less than points per second set

### DISCUSSION THREAD

**Discussion by akshay-antony** [carla-simulator/carla#4754] [Q&A]
I have set the number of points per second as 32,000 for a 16-channel lidar. But in a point cloud I am getting only 2000 points. Is it related to the frequency of rotation(20 in my case)? Or fps of data recording. I am using Carla-ros-bridge

**Comment by nparker2020** [comment]
I have a similar situation on my end. 

I am trying to simulate a 2D LiDAR with a 150Hz scan frequency, that provides 366 points-per-scan.
The LiDAR has a angular resolution of 0.75 degrees, and a horizontal FOV of 275 degrees.
So, the LiDAR sensor configuration is as follows:
upper_fov: 0.0
lower_fov: 0.0
channels: 1
rotation_frequency: 150
points_per_second: 55,500
horizontal_fov: 275

I set the fixed_delta_seconds configuration variable to 0.0066, which should give a 150 FPS figure.
However, the resulting refresh rate of the simulated sensor always has a maximum of 60FPS, and delivers less than 366 points. Coincidentally, my monitor is also locked to this refresh rate. My current theory is that Unreal, and therefore Carla, are bounded by the framerate of the monitor.

Is this the case? Has anyone noticed/experienced this also?

Would love to hear any solutions/workarounds for bumping sensor data frequency > 60Hz.
